server/container/guice/cassandra-guice/src/test/java/org/apache/james/ESReporterTest.java

```diff
@@ -0,0 +1,156 @@
+
+    private Timer timer;
+
+    @Rule
+    public CassandraJmapTestRule cassandraJmap = new CassandraJmapTestRule(embeddedCassandraRule, new EmbeddedCassandraRule());
+
+    private JmapJamesServer server;
+    private AccessToken accessToken;
+
+    @Before
+    public void setup() throws Exception {
+        server = cassandraJmap.jmapServer();
+        server.start();
+        GuiceServerProbe serverProbe = server.getGuiceProbeProvider().getProbe(GuiceServerProbe.class);
+        serverProbe.addDomain(DOMAIN);
+        serverProbe.addUser(USERNAME, PASSWORD);
+
+        RestAssured.requestSpecification = new RequestSpecBuilder()
+                .setContentType(ContentType.JSON)
+                .setAccept(ContentType.JSON)
+                .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8)))
+                .setPort(server.getJmapProbe().getJmapPort())
+                .build();
+        accessToken = HttpJmapAuthentication.authenticateJamesUser(baseUri(), USERNAME, PASSWORD);
+
+        timer = new Timer();
+    }
+
+    private URIBuilder baseUri() {
+        return new URIBuilder()
+            .setScheme("http")
+            .setHost("localhost")
+            .setPort(server.getJmapProbe().getJmapPort())
+            .setCharset(Charsets.UTF_8);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (server != null) {
+            server.stop();
+        }
+
+        timer.cancel();
+    }
+
+    @Test
+    public void timeMetricsShouldBeReportedWhenImapCommandsReceived() throws Exception {
+        IMAPClient client = new IMAPClient();
+        client.connect(InetAddress.getLocalHost(), IMAP_PORT);
+        client.login(USERNAME, PASSWORD);
+        
+        TimerTask timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                try {
+                    client.list("", "*");
+                } catch (IOException e) {
+                    throw Throwables.propagate(e);
+                }
+            }
+        };
+        timer.schedule(timerTask, DELAY_IN_MS, PERIOD_IN_MS);
+
+        await().atMost(Duration.TEN_MINUTES)
+            .until(() -> done());
+    }
+
+    @Test
+    public void timeMetricsShouldBeReportedWhenJmapRequestsReceived() throws Exception {
+        TimerTask timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                    given()
+                        .header("Authorization", accessToken.serialize())
+                        .body("[[\"getMailboxes\", {}, \"#0\"]]")
+                    .when()
+                        .post("/jmap")
+                    .then()
+                        .statusCode(200);
+            }
+        };
+        timer.schedule(timerTask, DELAY_IN_MS, PERIOD_IN_MS);
+
+        await().atMost(Duration.TEN_MINUTES)
+            .until(() -> done());
+    }
+
+    private boolean done() {
```

Comment:
I think the name is not really good, what about `checkMetricRecordedInElasticSearch`